public/pages/CreateDataStream/containers/CreateDataStream/CreateDataStream.tsx

```diff
@@ -33,14 +39,28 @@ export default class CreateDataStream extends Component<CreateDataStreamProps> {
     this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.DATA_STREAMS, lastBread]);
   }
 
+  updateDataSourcePropsInUrl() {
```

Comment:
Still, copy / paste found, can we extract such code into a component like UrlPersistentComponent?

Reply:
Wrote a common hook and moved the URL updation to higher order component.